packages/tgpu-cli/gen.mjs

```diff
@@ -136,4 +141,162 @@ function replaceWithAlias(type) {
     : type.name;
 }
 
+/**
+ * @param { import('wgsl_reflect').VariableInfo[][] } bindGroups
+ */
+function generateBindGroupLayouts(bindGroups) {
+  return bindGroups.length > 0
+    ? `/* bindGroupLayouts */
+${bindGroups
+  .flatMap(
+    (group, index) => `\
+export const layout${index} = tgpu.bindGroupLayout({
+  ${generateGroupLayout(group)}
+}).$forceIndex(${index});`,
+  )
+  .join('\n\n')}`
+    : '';
+}
+
+const RESOURCE_GENERATORS = [
+  generateUniformVariable,
+  generateStorageVariable,
+  generateTextureVariable,
+  generateSamplerVariable,
+  generateStorageTextureVariable,
+];
+
+const ACCESS_TYPES = {
+  read: 'readonly',
+  write: 'writeonly',
+  read_write: 'mutable',
+};
+
+const SAMPLE_TYPES = {
+  u32: 'uint',
+  i32: 'sint',
+  f32: 'float',
+};
+
+/**
+ * @param { import('wgsl_reflect').VariableInfo[] } group
+ */
+function generateGroupLayout(group) {
+  let emptyCount = 0;
+
+  return Array(group.length)
+    .fill(null)
+    .map((_, index) => group[index])
+    .map((variable) =>
+      variable
+        ? `${variable.name}: ${generateVariable(variable)},`
+        : `_${emptyCount++}: null,`,
```

Comment:
What do you think about embedding a comment into the generated code, like `// skipping binding N`?
Just so that the purpose of this field is apparent.
Also, we could simplify the empty indexing by breaking an requirement I made initially XD. We can just number them based on the binding index instead.

```suggestion
        : `_${index}: null, // skipping binding ${index}`,
```

Reply:
sure, i'll add a comment. and using index means we remove the only `let` in the file, which I'm fond of